Source/HyprMXAdapterRewardedAd.swift

```diff
@@ -18,21 +18,28 @@ final class HyprMXAdapterRewardedAd: HyprMXAdapterAd, PartnerAd {
     /// - parameter completion: Closure to be performed once the ad has been loaded.
     func load(with viewController: UIViewController?, completion: @escaping (Result<PartnerEventDetails, Error>) -> Void) {
         log(.loadStarted)
-        loadCompletion = completion
 
         // HyprMX only supports interaction from the Main Thread. Unlike banner ads, Chartboost Mediation SDK
         // does not load rewarded ads on the main thread so we need to wrap this ourselves.
         DispatchQueue.main.async { [self] in
             // Construct a partner ad to be persisted for subsequent ad operations.
             if let ad = HyprMX.getPlacement(self.request.partnerPlacement) {
                 self.ad = ad
-                ad.placementDelegate = self
-                ad.loadAd()
+                ad.expiredDelegate = self
+                ad.loadAd { success in
+                    if success {
+                        self.log(.loadSucceeded)
+                        completion(.success([:]))
+                    } else {
+                        let loadError = self.error(.loadFailureNoFill)
```

Comment:
Same here. I see the original code used a "no fill" error code, but I'm not sure it made sense.
```suggestion
                        let loadError = self.error(.loadFailureUnknown)
```

Reply:
I'll go ahead and commit both of your suggestions.  Makes better sense to just set them as unknown.